File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low = 1
        high = max(piles)


        while low <high:
            mid = (low+high)//2

            total_time = sum(math.ceil(p/mid) for p in piles)

            if total_time <=h:
                high = mid #try a smaller speed
            else:
                low = mid +1 
        return low 

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
  

        Arrays.sort(piles);
        int low = 1;
        int high= Arrays.stream(piles).max().getAsInt();

        while(low<high)
        {
            int mid= (low+high)/2;
            int total =0;
            for(int pile:piles)
            {
                total+=Math.ceil((double) pile/mid);
            }
            if(total<=h)
            {
                high = mid;
            }
            else 
            {
                low = mid+1;
            }
        }
        return low;
        
    }
}